Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -330,6 +490,150 @@ lemma iteratedFDeriv_fourierIntegral {N : ℕ∞}
+      exact (LinearIsometryEquiv.integrable_comp_iff _).1 h'f
+    suffices H : (fourierIntegral 𝐞 μ L.toLinearMap₂ (fderiv ℝ (iteratedFDeriv ℝ n f)) w)
+          (m 0) (Fin.tail m) =
+        (-(2 * π * I)) ^ (n + 1) • (∏ x : Fin (n + 1), -L (m x) w) • ∫ v, 𝐞 (-L v w) • f v ∂μ by
+      rw [fourierIntegral_continuousMultilinearMap_apply' (h'f _ hn)]
+      simp only [iteratedFDeriv_succ_apply_left, fourierPowSMulRight_apply,
+        ContinuousLinearMap.neg_apply, ContinuousLinearMap.flip_apply]
+      rw [← fourierIntegral_continuousMultilinearMap_apply' ((J.apply_continuousLinearMap _)),
+          ← fourierIntegral_continuousLinearMap_apply' J]
+      exact H
+    have h'n : n < N := lt_of_lt_of_le (by simp [-Nat.cast_add]) hn
+    rw [fourierIntegral_fderiv]
+    · have A : ∀ (x : ℝ) (v : E), x • v = (x : ℂ) • v := fun x v ↦ rfl
+      simp only [ih h'n.le, fourierSMulRight_apply, ContinuousLinearMap.neg_apply,
+        ContinuousLinearMap.flip_apply, neg_smul, smul_neg, neg_neg, smul_apply,
+        fourierPowSMulRight_apply, A, smul_smul]
+      congr 1
+      have B : ∀ (i : Fin n), Fin.tail m i = m (Fin.succ i) := fun i ↦ rfl
+      simp only [ofReal_prod, ofReal_neg, pow_succ, mul_neg, Fin.prod_univ_succ, neg_mul,
+        ofReal_mul, neg_neg, B]
+      ring
+    · exact h'f n h'n.le
+    · exact hf.differentiable_iteratedFDeriv h'n
+    · exact J
+
+/-- The `k`-th derivative of the Fourier integral of `f`, multiplied by `(L v w) ^ n`, is the
+Fourier integral of the `n`-th derivative of `(L v w) ^ k * f`. -/
+theorem fourierPowSMulRight_iteratedFDeriv_fourierIntegral [FiniteDimensional ℝ V]
+    {μ : Measure V} [Measure.IsAddHaarMeasure μ] {K N : ℕ∞} (hf : ContDiff ℝ N f)
+    (h'f : ∀ (k n : ℕ), k ≤ K → n ≤ N → Integrable (fun v ↦ ‖v‖^k * ‖iteratedFDeriv ℝ n f v‖) μ)
+    {k n : ℕ} (hk : k ≤ K) (hn : n ≤ N) {w : W} :
+    fourierPowSMulRight (-L.flip)
+      (iteratedFDeriv ℝ k (fourierIntegral 𝐞 μ L.toLinearMap₂ f)) w n =
+    fourierIntegral 𝐞 μ L.toLinearMap₂
+      (iteratedFDeriv ℝ n (fun v ↦ fourierPowSMulRight L f v k)) w := by
+  rw [fourierIntegral_iteratedFDeriv (N := N) _ (hf.fourierPowSMulRight _ _) _ hn]
+  · congr
+    rw [iteratedFDeriv_fourierIntegral (N := K) _ _ hf.continuous.aestronglyMeasurable hk]
+    intro k hk
+    simpa only [norm_iteratedFDeriv_zero] using h'f k 0 hk bot_le
+  · intro m hm
+    have I : Integrable (fun v ↦ ∑ p in Finset.range (k + 1) ×ˢ Finset.range (m + 1),
+        ‖v‖ ^ p.1 * ‖iteratedFDeriv ℝ p.2 f v‖) μ := by
+      apply integrable_finset_sum _ (fun p hp ↦ ?_)
+      simp only [Finset.mem_product, Finset.mem_range_succ_iff] at hp
+      apply h'f _ _ ((Nat.cast_le.2 hp.1).trans hk) ((Nat.cast_le.2 hp.2).trans hm)
+    apply (I.const_mul ((2 * π) ^ k * (2 * k + 2) ^ m * ‖L‖ ^ k)).mono'
+      ((hf.fourierPowSMulRight L k).continuous_iteratedFDeriv hm).aestronglyMeasurable
+    filter_upwards with v
+    apply norm_iteratedFDeriv_fourierPowSMulRight _ hf hm _
+    intro i hi j hj
```

Comment:
```suggestion
    refine norm_iteratedFDeriv_fourierPowSMulRight _ hf hm (fun i hi j hj ↦ ?_)
```
(I'm not so fond of `apply` with multiple `_`'s, for the same reason as lean3 `refine` -- it's not obvious which blanks are to be synthesized and which to be left as new goals)